examples/jetstream/pull-consumer/c/main.c

```diff
@@ -0,0 +1,345 @@
+    // are fetching pre-buffered messages. The response time is in single ms.
+    //
+    // **Note**: each fetched message must be acknowledged.
+    //
+    // **Note**: `natsMsgList_Destroy` will destroy the fetched messages.
+    for (ibatch = 0, c = 0; (s == NATS_OK) && (c < NUM_MESSAGES); ibatch++)
+    {
+        int64_t start = nats_Now();
+        s = natsSubscription_Fetch(&list, sub, 2, 5000, &jerr);
+        if (s == NATS_OK)
+        {
+            c += (int64_t)list.Count;
+            printf("exampleFetch: batch #%d (%d messages) in %dms\n", ibatch, list.Count, (int)(nats_Now() - start));
+        }
+        else
+        {
+            printf("exampleFetch: error: %d:\n", s);
+            nats_PrintLastErrorStack(stderr);
+        }
+        for (i = 0; (s == NATS_OK) && (i < list.Count); i++)
+        {
+            s = natsMsg_Ack(list.Msgs[i], jsOpts);
+            printf("exampleFetch: received and acked message on %s\n", natsMsg_GetSubject(list.Msgs[i]));
+        }
+        natsMsgList_Destroy(&list);
+    }
+
+    // Attempt to fetch more messages, but this time we will wait for the
+    // 500ms timeout since there are no more pre-buffered messages.
+    if (s == NATS_OK)
+    {
+        int64_t start = nats_Now();
+        s = natsSubscription_Fetch(&list, sub, 2, 500, &jerr);
+        printf("exampleFetch: extra natsSubscription_Fetch returned status %d and %d messages in %dms\n",
+               s, list.Count, (int)(nats_Now() - start));
+        s = NATS_OK;
+    }
+
+    // Cleanup.
+    natsSubscription_Destroy(sub);
+
+    return s;
+}
+
+// Create another similar pull consumer subscription and use
+// `natsSubscription_FetchRequest` to receive messages with more precise
+// control.
+static natsStatus exampleFetchRequest(jsCtx *js, jsOptions *jsOpts)
+{
+    natsStatus s = NATS_OK;
+    natsSubscription *sub = NULL;
+    jsErrCode jerr = 0;
+    natsMsgList list = {0};
+    jsSubOptions so;
+    int c, i, ibatch;
+
+    // Create another similar pull consumer, same scenario but using
+    // `natsSubscription_FetchRequest` for precise control.
+    s = jsSubOptions_Init(&so);
+    if (s == NATS_OK)
+    {
+        printf("exampleFetchRequest: create pull consumer and use natsSubscription_FetchRequest to receive messages\n");
+        so.Stream = STREAM_NAME;
+        s = js_PullSubscribe(&sub, js, SUBSCRIBE_SUBJECT, NULL, jsOpts, &so, &jerr);
+    }
+
+    // Use `natsSubscription_FetchRequest` to fetch the messages.
+    //
+    // We set the batch size to 1000, but MaxBytes of 300 so we will only get 2
+    // messages at a time. Note that since we do not set NoWait, the call will
+    // block until the batch is filled or the timeout expires, unlike
+    // `natsSubscription_Fetch`.
+    for (ibatch = 0, c = 0; (s == NATS_OK) && (c < NUM_MESSAGES); ibatch++)
+    {
+        int64_t start = nats_Now();
+        jsFetchRequest fr = {
+            .Batch = 1000,
+            // .NoWait = true,
```

Comment:
Suggest removing this since it shows up as text when the example renders.

Reply:
Updated to a /**/ with a note.